optimum/intel/openvino/modeling_sam.py

```diff
@@ -0,0 +1,365 @@
+        )
+        vision_encoder_file_name = vision_encoder_file_name or default_vision_encoder_file_name
+        prompt_encoder_mask_decoder_file_name = (
+            prompt_encoder_mask_decoder_file_name or default_prompt_encoder_mask_decoder_file_name
+        )
+
+        model_file_names = {
+            "vision_encoder_model": vision_encoder_file_name,
+            "prompt_encoder_mask_decoder_model": prompt_encoder_mask_decoder_file_name,
+        }
+
+        if not from_onnx:
+            model_file_names.update(
+                {
+                    "vision_encoder_model_bin": vision_encoder_file_name.replace(".xml", ".bin"),
+                    "prompt_encoder_mask_decoder_model_bin": prompt_encoder_mask_decoder_file_name.replace(
+                        ".xml", ".bin"
+                    ),
+                }
+            )
+
+        compile_only = kwargs.get("compile_only", False)
+        if os.path.isdir(model_id):
+            # Load model from a local directory
+            model_save_dir = Path(model_id)
+            file_names = {k: os.path.join(model_id, model_file_names[k]) for k in model_file_names}
+        else:
+            file_names = {}
+            for name, file_name in model_file_names.items():
+                model_cache_path = cls._cached_file(
+                    repo_id=model_id,
+                    filename=file_name,
+                    token=token,
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    local_files_only=local_files_only,
+                )
+                file_names[name] = model_cache_path
+            model_save_dir = Path(model_cache_path).parent
+        if not compile_only:
+            vision_encoder_model = cls.load_model(file_names["vision_encoder_model"])
+            prompt_encoder_model = cls.load_model(file_names["prompt_encoder_mask_decoder_model"])
+        else:
+            vision_encoder_model = cls._compile_model(
+                file_names["vision_encoder_model"],
+                kwargs.get("device", "CPU"),
+                kwargs.get("ov_config"),
+                model_save_dir,
+            )
+            prompt_encoder_model = cls._compile_model(
+                file_names["prompt_encoder_mask_decoder_model"],
+                kwargs.get("device", "CPU"),
+                kwargs.get("ov_config"),
+                model_save_dir,
+            )
+
+        model = cls(
+            vision_encoder_model=vision_encoder_model,
+            prompt_encoder_mask_decoder_model=prompt_encoder_model,
+            config=config,
+            model_save_dir=model_save_dir,
+            **kwargs,
+        )
+
+        return model
+
+    @property
+    def _ov_submodel_names(self):
+        model_names = ["vision_encoder_model", "prompt_encoder_mask_decoder_model"]
+        return model_names
+
+    def reshape(self, batch_size: int, sequence_length: int):
```

Comment:
let's log a warning that it's not yet supported

Reply:
I provided reshape support